### PR TITLE
Generate wrappers for ReactNativeFlows that properly serialize complex parameters

### PR DESCRIPTION
**Pull request by erksch**
```kotlin
@ReactNativeModule("MyRNModule")
class MyRnModule() {
  @ReactNativeFlow
  fun myFlow(myParam: List<String>): Flow<Int> = ...
}
```

Before

```typescript
interface MyRNModuleInterface {
  myFlow: Next1<number, string[]>;
}

const MyRNModule: MyRNModuleInterface = {
  ...NativeModules.MyRnModule 
  // no wrappers for myFlow
}
```

After

```typescript
const MyRNModule: MyRNModuleInterface = {
  ...NativeModules.MyRNModule,
  myFlow: (currentValue: string | null, arg: string[]) => NativeModules.MyRNModule.myFlow(
     currentValue, 
     JSON.stringify(arg)
  ),
}
```

### CODE REVIEW

**Review by Legion2 — CHANGES_REQUESTED** [review]
Please add more test cases that cover the change

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/E2ETest.kt [hunk 1]
```diff
@@ -103,7 +103,17 @@ class E2ETest {
...
+        return flowOf()
+    }
+
+    @ReactNativeFlow
+    fun testFlowParameterized2(arg1: Int, arg2: String): Flow<FlowTest> {
+        return flowOf()
+    }
+
+    @ReactNativeFlow
+    fun testFlowParameterizedMany(arg1: Int, arg2: String, arg3: List<String>): Flow<FlowTest> {
```
[the remark this answers is not in the document]
Comment (erksch) [reply example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/E2ETest.kt]: done :+1:

**Comment by erksch** [comment]
I am releasing this as a patch release. Because it does not break existing primitive argument flows and fixes broken complex flows.

File: example/src/generated/modules.ts [hunk 2]
```diff
@@ -12,7 +12,15 @@ interface NativeE2ETestInterface {
 
   testFlowComplex: Next<com.myrnproject.shared.Test>;
 
-  testFlowType: Next1<com.myrnproject.shared.FlowTest, number>;
+  testFlowParameterized: Next1<com.myrnproject.shared.FlowTest, number>;
+
+  testFlowParameterized2: Next2<com.myrnproject.shared.FlowTest, number, string>;
+
+  testFlowParameterizedComplex: Next1<com.myrnproject.shared.FlowTest, com.myrnproject.shared.Test>;
```
Comment (Legion2) [inline example/src/generated/modules.ts:19]: Shouldn't this be string as the second argument, because this expects the argument to already be serialized?

Comment (erksch) [reply example/src/generated/modules.ts:19]: No this are the types from the perspective of the caller of useFlow who wants to pass his complex argument as is, not as a seralized string.

Comment (Legion2) [reply example/src/generated/modules.ts:19]: No this is the native view which is already serialized, see the interface name. Below is the caller facing interface.

Comment (erksch) [reply example/src/generated/modules.ts:19]: Oh I see! This is a whole nother issue! I'll fix it

Comment (erksch) [reply example/src/generated/modules.ts]: fixed it

**Review by Legion2 — APPROVED** [review]
Nice work

**Pull request merged**